ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/PeerDenylistManager.java

```diff
@@ -46,8 +50,12 @@ public void onDisconnect(
       final DisconnectReason reason,
       final boolean initiatedByPeer) {
     if (shouldBlock(reason, initiatedByPeer)) {
-      LOG.debug("Added peer {} to peer denylist for reason {}", connection, reason.name());
-      denylist.add(connection.getPeer());
+      if (maintainedPeers.contains(connection.getPeer())) {
+        LOG.debug("Skip adding maintained peer {} to denylist for reason {}", connection, reason.name());
```

Comment:
the other log message has "to _peer_ denylist", not "to denylist". Or just different wording in general, as this is a bit awkward.

Reply:
I don't see anywhere that maintainedPeers gets called outside of a test class. It is possible this is dead code. You seem to be under the impression that this is where static peers are maintained, which seems reasonable to me... I just don't see any way to prove it.